lib/azure/armrest/configuration.rb

```diff
@@ -108,23 +110,26 @@ def initialize(args)
 
         options.each { |key, value| send("#{key}=", value) }
 
-        unless client_id && client_key && tenant_id && subscription_id
-          raise ArgumentError, "client_id, client_key, tenant_id and subscription_id must all be specified"
+        unless client_id && client_key && tenant_id
+          raise ArgumentError, "client_id, client_key, and tenant_id must all be specified"
         end
 
         # Allows for URI objects or Strings.
         @proxy = @proxy.to_s if @proxy
```

Comment:
They are mostly set as accessors so that I can easily use hash args + send. I agree that this is probably not ideal but would like to keep that a separate PR.